services/ark-api/ark-api/src/ark_api/api/v1/openai.py

```diff
@@ -177,39 +121,41 @@ async def chat_completions(request: ChatCompletionRequest) -> ChatCompletion:
                     ark_client, query_name, model, input_text
                 )
 
-            # Streaming was requested - we'll check to see if the backend is
-            # configured to support streaming, and if so its streaming endpoint.
-            has_streaming, streaming_url = await check_streaming_availability(ark_client, query_name, "default")
-
-            # Regardless of what we return, i'll be a streaming response with
-            # SSE headers.
-            sse_headers = {
-                "Cache-Control": "no-cache",
-                "Connection": "keep-alive",
-            }
-
-            # If the backend has streaming and we have an endpoint, proxy the
-            # endpoint to the caller, this gives true real-time streaming.
-            if has_streaming and streaming_url:
-                logger.info(f"Streaming available for query: {query_name}")
+            # Streaming was requested - check if streaming backend is available
+            api = k8s_client.ApiClient()
+            v1 = k8s_client.CoreV1Api(api)
+            streaming_config = await get_streaming_config(v1, namespace)
+
+            # If no config or not enabled, fall back to polling
+            if not streaming_config or not streaming_config.enabled:
+                logger.info("No streaming backend configured, falling back to polling")
+                completion = await poll_query_completion(
+                    ark_client, query_name, model, input_text
+                )
+                sse_lines = create_single_chunk_sse_response(completion)
                 return StreamingResponse(
-                    proxy_streaming_response(streaming_url),
+                    iter(sse_lines),
                     media_type="text/event-stream",
-                    headers=sse_headers
+                    headers={
```

Comment:
Define these earlier in the function so we can use `sseHeaders` and when we define them clearly state what they are for